### PR TITLE
Remove IGraphServiceClient interface and add TokenCredential constructor(feature/2.0)

### PR DESCRIPTION
**Pull request by andrueastman**
This PR is part of the work on the generator to 
- Remove IGraphServiceClient interface 
- Add TokenCredential constructor

See https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/477

### CODE REVIEW

File: src/Microsoft.Graph/Generated/requests/GraphServiceClient.cs [hunk 1]
```diff
@@ -29,6 +31,20 @@ public partial class GraphServiceClient : BaseClient, IGraphServiceClient
...
+        /// Instantiates a new GraphServiceClient using a <see cref="TokenCredential"/> instance.
+        /// </summary>
+        /// <param name="tokenCredential">The <see cref="TokenCredential"/> to use for authentication</param>
+        /// <param name="scopes">Scopes required to access Microsoft Graph. This defaults to https://graph.microsoft.com/.default when none is set.</param>
+        /// <param name="httpProvider">The <see cref="IHttpProvider"/> for sending requests.</param>
+        public GraphServiceClient(
+            TokenCredential tokenCredential,
+            IEnumerable<string> scopes = null,
+            IHttpProvider httpProvider = null)
+            : this("https://graph.microsoft.com/v1.0", new TokenCredentialAuthProvider(tokenCredential,scopes), httpProvider)
```
Comment (baywet) [inline src/Microsoft.Graph/Generated/requests/GraphServiceClient.cs:44]: https://github.com/microsoftgraph/msgraph-beta-sdk-dotnet/pull/244#discussion_r591473350

**Review by baywet — COMMENTED** [review]
See the comment I left in beta

**Review by MIchaelMainer — APPROVED** [review]
approved with a +1 on Vincent's suggested change.

File: src/Microsoft.Graph/Generated/requests/GraphServiceClient.cs (hunk 1 above)
Comment (andrueastman) [reply src/Microsoft.Graph/Generated/requests/GraphServiceClient.cs:44]: The Auth provider will set the scopes to be a default value if not provided therefore no need to throw. See [here](https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/blob/bcbc09cbe9b5e4be80f3a39acbd1cd5361257a81/src/Microsoft.Graph.Core/Authentication/TokenCredentialAuthProvider.cs#L36).
Furthermore, the XML comments of the constructor capture this as well.

**Pull request merged**